poly-commit/src/linear_codes/mod.rs

```diff
@@ -339,37 +335,24 @@ where
             // 3. Generate vector `b` to left-multiply the matrix.
             let (_, b) = L::tensor(point, n_cols, n_rows);
 
-            let mut transcript = IOPTranscript::new(b"transcript");
-            transcript
-                .append_serializable_element(b"root", root)
-                .map_err(|_| Error::TranscriptError)?;
+            sponge.absorb(&to_bytes!(&commitment.root).map_err(|_| Error::TranscriptError)?);
 
             // If we are checking well-formedness, we need to compute the well-formedness proof (which is just r.M) and append it to the transcript.
             let well_formedness = if ck.check_well_formedness() {
-                let mut r = Vec::new();
-                for _ in 0..n_rows {
-                    r.push(
-                        transcript
-                            .get_and_append_challenge(b"r")
-                            .map_err(|_| Error::TranscriptError)?,
-                    );
-                }
+                // TODO Check this: we changed, now we squeeze all elements and then abosrb all of them
+                let r = sponge.squeeze_field_elements::<F>(n_rows);
+                sponge.absorb(&r);
                 let v = mat.row_mul(&r);
 
-                transcript
-                    .append_serializable_element(b"v", &v)
-                    .map_err(|_| Error::TranscriptError)?;
+                sponge.absorb(&v);
                 Some(v)
             } else {
                 None
             };
 
             let point_vec = L::point_to_vec(point.clone());
-            for element in point_vec.iter() {
-                transcript
-                    .append_serializable_element(b"point", element)
-                    .map_err(|_| Error::TranscriptError)?;
-            }
+            // TODO Why not absorbing the vector all at once?
```

Comment:
The TODO is no longer needed right?

Reply:
So, previously we iterated over the vector and appended each item to the transcript. I wonder why we didn't append the entire vector. That's what I did here, and I wanted to double-check with you if it is okay.

Reply:
Thanks! Yes, the verification is symmetric to this part too.